src/ol/source/tileutfgridsource.js

```diff
@@ -49,7 +49,13 @@ ol.source.TileUTFGrid = function(options) {
    */
   this.template_ = undefined;
 
-  ol.net.jsonp(options.url, this.handleTileJSONResponse.bind(this));
+  goog.asserts.assert(options.url || options.tileJSON,
+      'Either url or tileJSON options must be provided');
+  if (options.url) {
+    ol.net.jsonp(options.url, this.handleTileJSONResponse.bind(this));
+  } else if (options.tileJSON) {
+    this.handleTileJSONResponse(options.tileJSON);
+  }
```

Comment:
super minor but instead of the `goog.asserts.assert` above, an else branch can be added:
```js
if (options.url) {
    ol.net.jsonp(options.url, this.handleTileJSONResponse.bind(this));
  } else if (options.tileJSON) {
    this.handleTileJSONResponse(options.tileJSON);
  } else {
    goog.asserts.fail('Either url or tileJSON options must be provided')
  }

```